res/controllers/Reloop-Mixage.scripts.js

```diff
@@ -0,0 +1,358 @@
+            } else {
+                engine.setValue("[PreviewDeck1]", "LoadSelectedTrackAndPlay", true);
+            }
+        } else if (status === 0xB0) { // "push2browse" button was turned
+            var newValue = value - 64;
+            Mixage.scrollLibrary(newValue);
+        }
+    } else if (control === 0x0D || control === 0x4C) { // load into deck 1
+        if (value === 0x7F) {
+            engine.setValue("[PreviewDeck1]", "stop", true);
+            engine.setValue("[Channel1]", control === 0x4C ? "LoadSelectedTrackAndPlay" : "LoadSelectedTrack", true);
+            Mixage.libraryRemainingTime = Mixage.libraryReducedHideTimeout;
+        }
+    } else if (control === 0x1B || control === 0x5A) { // load into deck 2
+        if (value === 0x7F) {
+            engine.setValue("[PreviewDeck1]", "stop", true);
+            engine.setValue("[Channel2]", control === 0x5A ? "LoadSelectedTrackAndPlay" : "LoadSelectedTrack", true);
+            Mixage.libraryRemainingTime = Mixage.libraryReducedHideTimeout;
+        }
+    }
+};
+
+// Set the library visible and hide it when libraryHideTimeOut is reached
+Mixage.setLibraryMaximized = function(visible) {
+    if (visible === true) {
+        Mixage.libraryRemainingTime = Mixage.libraryHideTimeout;
+        // maximize library if not maximized already
+        if (engine.getValue("[Master]", "maximize_library") !== true) {
+            engine.setValue("[Master]", "maximize_library", true);
+            if (Mixage.libraryHideTimer === 0) {
+                // timer not running. start it
+                Mixage.libraryHideTimer = engine.beginTimer(Mixage.libraryHideTimeout / 5, Mixage.libraryCheckTimeout);
+            }
+        }
+    } else {
+        if (Mixage.libraryHideTimer !== 0) {
+            engine.stopTimer(Mixage.libraryHideTimer);
+            Mixage.libraryHideTimer = 0;
+        }
+        Mixage.libraryRemainingTime = 0;
+        engine.setValue("[Master]", "maximize_library", false);
+    }
+};
+
+Mixage.libraryCheckTimeout = function() {
+    Mixage.libraryRemainingTime -= Mixage.libraryHideTimeout / 5;
+    if (Mixage.libraryRemainingTime <= 0) {
+        engine.stopTimer(Mixage.libraryHideTimer);
+        Mixage.libraryHideTimer = 0;
+        Mixage.libraryRemainingTime = 0;
+        engine.setValue("[Master]", "maximize_library", false);
+    }
+};
+
+// The "record" button that enables/disables scratching
+Mixage.scratchActive = function(channel, control, value/*, status, group*/) {
+    // calculate deck number from MIDI control. 0x04 controls deck 1, 0x12 deck 2
+    var deckNr = control === 0x04 ? 1 : 2;
+    if (value === 0x7F) {
+        Mixage.scratchPressed = true;
+        var alpha = 1.0 / 8.0;
+        var beta = alpha / 32.0;
+        engine.scratchEnable(deckNr, 620, 33.0 + 1.0 / 3.0, alpha, beta);
+    } else {
+        Mixage.scratchPressed = false;
+        engine.scratchDisable(deckNr);
```

Comment:
this would instantly slow down the deck as if someone had touched the platter, is that really desired? The canonical implementation would just toggle `Mixage.scratchPressed` so when you actually touch the well, you can differentiate between jog and stratching. See https://github.com/mixxxdj/mixxx/wiki/Components-JS#jogwheelbasic

Reply:
JogWheelBasic looks good, but I don't have the 2.3.4, so I'll use the old-style scratching. Will fix this though.